src/solvers/refinement/string_constraint_generator_main.cpp

```diff
@@ -272,6 +272,37 @@ string_exprt string_constraint_generatort::add_axioms_for_if(
   return res;
 }
 
+/// Add axioms enforcing that the content of the first array is equal to
+/// the true case array if the condition is true and
+/// the else case array otherwise.
+/// \param lhs: an expression
+/// \param expr: an if expression of type array
+void string_constraint_generatort::add_axioms_for_if_array(
+  const exprt &lhs, const if_exprt &expr)
+{
+  PRECONDITION(lhs.type()==expr.type());
```

Comment:
There is no namespace available from here, is it worth including it as a class member?